src/PermissionManager.sol

```diff
@@ -86,6 +95,11 @@ contract PermissionManager is IERC1271, Ownable, Pausable {
     /// @dev Storage not keyable by account, can only be accessed in execution phase.
     mapping(address permissionContract => bool enabled) internal _enabledPermissionContracts;
 
+    /// @notice Track if a permission contract should account for gas spent by paymaster.
+    ///
+    /// @dev Storage not keyable by account, can only be accessed in execution phase.
+    mapping(address paymaster => bool enabled) public addPaymasterGasSpend;
```

Comment:
nit: I find this name here and in the event and error kind confusing. Maybe like `shouldAddPaymasterGasToTotalSpend`